### PR TITLE
[universal] - check toolings - monthly task

### PR DESCRIPTION
**Pull request by gauravsaini04**
**Dev container name**:
 
 * Universal
 
 **Description**:
 
 This PR updates the toolings versions for `Universal` dev container image to `long term supported` versions if available for the runtime, else uses the `latest` versions of those runtimes
 
 _Changelog_:
 
 * Updated `devcontainer.json` file
 
 **Checklist**:
 
 * [x]   Checked that applied changes work as expected

### CODE REVIEW

**Comment by khgi6565** [comment]
> **Dev container name**:
> 
> * Universal
> 
> **Description**:
> 
> This PR updates the toolings versions for `Universal` dev container image to `long term supported` versions if available for the runtime, else uses the `latest` versions of those runtimes
> 
> _Changelog_:
> 
> * Updated `devcontainer.json` file
> 
> **Checklist**:
> 
> * [x]    Checked that applied changes work as expected

File: src/universal/.devcontainer/devcontainer.json [hunk 1]
```diff
@@ -23,8 +23,8 @@
         },
         "./local-features/nvs": "latest",
         "ghcr.io/devcontainers/features/python:1": {
-            "version": "3.10.13",
-            "additionalVersions": "3.9.18",
+            "version": "3.12.3",
```
Comment (samruddhikhandale) [inline src/universal/.devcontainer/devcontainer.json:26]: Oryx has compatibility issues with the version of Python installed, hence can we update the version based on the closest lts version which is supported by oryx?

Looking at https://github.com/devcontainers/images/actions/runs/9771395221/job/26974077297?pr=1108#step:3:38957 we'd need `3.12.1`

image

Comment (samruddhikhandale) [reply src/universal/.devcontainer/devcontainer.json:26]: That's the reason why we have tests to ensure the installed version is compatible with Oryx

File: src/universal/.devcontainer/local-features/patch-python/install.sh [hunk 2]
```diff
@@ -38,13 +38,3 @@ update_package() {
     sudo_if "$PYTHON_PATH -m pip install --upgrade --no-cache-dir $PACKAGE==$VERSION"
     sudo_if "$PYTHON_PATH -m pip show --no-python-version-warning $PACKAGE"
 }
-
-# Temporary: Upgrade python packages due to security vulnerabilities
-# They are installed by the base image (python) which does not have the patch.
-
-# https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-40897
```
Comment (samruddhikhandale) [inline src/universal/.devcontainer/local-features/patch-python/install.sh:45]: Niceeee

**Review by samruddhikhandale — CHANGES_REQUESTED** [review]

File: src/universal/.devcontainer/devcontainer.json (hunk 1 above)
Comment (gauravsaini04) [reply src/universal/.devcontainer/devcontainer.json]: thanks for info, done

File: src/universal/.devcontainer/devcontainer.json [hunk 3]
```diff
@@ -39,8 +39,8 @@
             "version": "latest"
         },
         "ghcr.io/devcontainers/features/ruby:1": {
-            "version": "3.2",
-            "additionalVersions": "3.1"
+            "version": "3.3.3",
```
Comment (samruddhikhandale) [inline src/universal/.devcontainer/devcontainer.json:42]:
```suggestion
            "version": "3.3.4",
```

Can we update to 3.3.4?  Hoping to not run into a similar issue as https://github.com/devcontainers/images/issues/1123

**Review by samruddhikhandale — CHANGES_REQUESTED** [review]

**Review by samruddhikhandale — APPROVED** [review]

**Pull request merged**